File: project/src/main/java/iteration2/project/FteCourse.java
package iteration2.project;

import java.util.HashMap;

public class FteCourse extends Course{

	public FteCourse(int courseSemester, String courseCode, String courseName, int credit, int akts, int quota, String courseType) {
		super(courseSemester, courseCode, courseName, credit, akts, quota, courseType);
	}
	
	public boolean checkRequirement(Student student, String semester) {
		int noOfFTECourses = 0;
		for(Course course: student.getRequestedCourses()) {
			if(course.getCourseCode().equals(this.getCourseCode())) continue;
			if(course instanceof FteCourse) {
				noOfFTECourses++;
			}
			
		}
		
		// student can not take FTE courses more than 1.
		if(noOfFTECourses >= 1) return false;
		
		return true;
		
	}
	

}

File: py_iteration/FteCourse.py
from Course import Course


class FteCourse(Course):
    def __init__(self, course_semester, course_code, course_name, credit, akts, quota, course_type):
        super().__init__(course_semester, course_code, course_name, credit, akts, quota, course_type)

    def check_course_requirement(self, **kwargs):
        accepted_courses = kwargs['accepted_courses']
        no_of_fte_courses = 0
        for course in accepted_courses:
            if type(course).__name__ == 'FteCourse':
                no_of_fte_courses += 1

        if no_of_fte_courses > 1:
            return False, 'can not take FTE courses more than 1.'
        return True, None